cli_tools/gce_windows_upgrade/upgrader/upgrader.go

```diff
@@ -0,0 +1,234 @@
+package upgrader
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"regexp"
+	"strings"
+
+	daisyutils "github.com/GoogleCloudPlatform/compute-image-tools/cli_tools/common/utils/daisy"
+	"github.com/GoogleCloudPlatform/compute-image-tools/cli_tools/common/utils/path"
+	"github.com/GoogleCloudPlatform/compute-image-tools/cli_tools/daisycommon"
+	"github.com/GoogleCloudPlatform/compute-image-tools/daisy"
+	daisyCompute "github.com/GoogleCloudPlatform/compute-image-tools/daisy/compute"
+	"google.golang.org/api/option"
+)
+
+const (
+	logPrefix                      = "[windows-upgrade]"
+	rebootWorkflowPath =             "daisy_workflows/windows_upgrade/reboot.wf.json"
+	upgradePreparationWorkflowPath = "daisy_workflows/windows_upgrade/windows_upgrade_preparation.wf.json"
+	upgradeWorkflowPath            = "daisy_workflows/windows_upgrade/windows_upgrade_2008r2_to_2012r2.wf.json"
+	retryUpgradeWorkflowPath       = "daisy_workflows/windows_upgrade/windows_upgrade_2008r2_to_2012r2_retry.wf.json"
```

Comment:
We should parametrize upgrade paths to easily support future paths (e.g. 2012->2016).